Library/Homebrew/missing_formula.rb

```diff
@@ -1,11 +1,17 @@
 require "formulary"
+require "cask/cmd/abstract_command"
+require "cask/cmd/info"
+require "cask/cask_loader"
+require "cask/installer"
 
 module Homebrew
   module MissingFormula
     class << self
       def reason(name, silent: false)
-        blacklisted_reason(name) || tap_migration_reason(name) ||
-          deleted_reason(name, silent: silent)
+        search_for_cask(name)
```

Comment:
The API for this is e.g. `MissingFormula.reason(query, silent: true)` so it will need to return a string with the Cask info or `nil` if the cask is not found. If there's any output at all it should be silenced if `silent: true` is passed.